### DISCUSSION TITLE
How to fix error code 119 in FileStation upload

### DISCUSSION THREAD

**Discussion by dMAC95** [N4S4/synology-api#132] [Show and tell]
I've been banging my head against the wall for hours trying to fix this issue. I don't know if this is really the right place for this, but I need to share it somewhere, to save someone else from this headache.

I couldn't get a file to upload, it kept giving me error code 119 and this was the request I was using to post the data:

```
def upload_file_stream(self):
        if not self.api_key:
            return None

        url = f"http://{self.nas_ip}:5000/webapi/entry.cgi"
        params = {
            "api": "SYNO.FileStation.Upload",
            "version": 2,
            "method": "upload",
            "_sid": self.api_key,
            "path": "/my_share/MAIN",
            "create_parents": "true",
            "overwrite":"true"
        }

        # Upload the file
        response = requests.post(url, data=params, files={'file':('test.txt',b'test','multipart/form-data')})
        print(response.text)
        result = response.json()

        if result.get("success", False):
            return True
        else:
            return False
```

I was so frustrated and tried every code combination I could think of. Even chat GPT gave up on it and told me to just contact support.

The trick that worked was to add the params as params assignment as well as the data assignment. I don't know why I didn't try that sooner because my other routes use that lol.

```
response = requests.post(url, params=params, data=params, files={'file':('test.txt',b'test','multipart/form-data')})
```

**Comment by dMAC95** [comment]
The code I wrote isn't derived from this code base, I was using this code base for reference.

**Comment by N4S4** [comment]
Hi, 
It is the right place and its ok, it is a great hint! 
I never had proroblems as long the file size is not bigger than the ram, I been working on it trying to chunk but no way. 
I will try to replicate your issue will see what comes out

**Reply by dMAC95** [reply]
Thanks!

**Comment by N4S4** [comment]
closed as a fix was implemented
